.github/workflows/test-python.yml

```diff
@@ -0,0 +1,48 @@
+name: test_python
+on:
+  push:
+  pull_request:
+
+env:
+  # build docker images to the (host) local registry
+  REGISTRY_LOCAL_ADDR: localhost:5000
+  # k8x requires the kind container address alias (comes from the kind-action)
+  REGISTRY_TEST_ADDR: kind-registry:5000
+  # If we are on develop or main, use the production repository
+  REGISTRY_PROD_ADDR: ghcr.io
+  # image name should be prefixed with the repository name
+  IMAGE_NAME: ${{ github.repository }}/cloudzero-agent-validator
+  
+jobs:
+  # This job detects if there are any changes to allow condition testing in other jobs
+  has_changes:
+    uses: ./.github/workflows/change-detector.yml
+  
+  test_python:
+    runs-on: ubuntu-latest
+    needs: has_changes
+    permissions:
+      contents: read
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```

Comment:
do we need to test all these other versions given that we're running in a container in which we control the python version?

also, maybe this is overkill, but we could also probably run the tests in the validator container image itself if we wanted to make sure we were always testing with the same python version

Reply:
I don't think so. I am paranoid since it is public. The image is using 3.12 - I will update the matrix to use only that version 🙌